### PR TITLE
Github Auth errors are logged

### LINKED ISSUE #64: Add error message when the github login fails
Hi, 

I am still having issues making an Heroky instance of mention-bot work with my private repos.

Webhook seems to be transmitted successfully but mention-bot keeps printing "Skipping because there are no reviewers found" even when I modify files where there should be reviewers.

Here is the only log I have (actual URLs, IDs and IPs has been faked for privacy reasons):

> 2015-12-16T16:50:40.702001+00:00 heroku[router]: at=info method=POST path="/" host=my-mention-bot.herokuapp.com request_id=ae90ffe6-d885-4e67-a9a5-f1365456c004 fwd="192.30.232.41" dyno=web.1 connect=1ms service=1008ms status=200 bytes=117
> 2015-12-16T16:50:40.690816+00:00 app[web.1]: https://github.com/my/project/pull/10 []
> 2015-12-16T16:50:40.690866+00:00 app[web.1]: Skipping because there are no reviewers found.

I am not yet accustomed with both NodeJS and Heroku :baby:. Appart from reading those log lines, what is the simplest way for me to debug my problem?

### PR DESCRIPTION
**Pull request by udnisap**
Fixes #64

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -56,10 +56,15 @@ var github = new GitHubApi({
...
-github.authenticate({
-  type: 'oauth',
-  token: process.env.GITHUB_TOKEN
-});
+try{
+  github.authenticate({
+    type: 'oauth',
+    token: process.env.GITHUB_TOKEN
+  });
+}catch(e){
```
Comment (vjeux) [inline server.js]: Nit: can you put a space before catch and after the closing parenthesis.

```
} catch(e) {

File: server.js [hunk 2]
```diff
@@ -56,10 +56,15 @@ var github = new GitHubApi({
   port: config.ghePort || '443'
 });
 
-github.authenticate({
-  type: 'oauth',
-  token: process.env.GITHUB_TOKEN
-});
+try{
```
Comment (vjeux) [inline server.js]:
```
try {

**Comment by vjeux** [comment]
Thanks! If you fix the two small nits i'll merge it in :)

**Comment by vjeux** [comment]
Hmmm actually did you try it out?

I don't think that it works. Authentication is asynchronous so it wouldn't throw synchronously. Also, if it actually threw an error then the entire node process would be killed with the correct error. But that's not what was described in the issue.

**Comment by jgardella** [comment]
@vjeux Actually I think authentication is [the one function in the GitHub API that is synchronous](https://github.com/mikedeboer/node-github#authentication).

**Comment by udnisap** [comment]
@vjeux I thought it was straightforward and didnt actually test.
@jgardella it is synchronous because its not making any api calls.  https://github.com/mikedeboer/node-github/blob/master/index.js#L417 I also looked at that doc and came up with this implementation. 

The actually issue is in the AuthCookie

**Comment by udnisap** [comment]
Should we just log or halt the application running by throwing an error?

**Comment by vjeux** [comment]
Either way is fine for me. Both are not going to work and the person will have to look the logs anyway to figure out what happened

**Comment by miketheman** [comment]
Hey gang, this is great, I now know that I'm getting auth errors.

I'm using a [Machine User](https://developer.github.com/guides/managing-deploy-keys/#machine-users) account - it has a username, password, token, and 2-factor auth enabled - so I'm not certain what combination of codes to use to get into private repos, nor how to debug it further.
Any advice?

**Pull request merged**